### PR TITLE
SOL-152495: Improving cut-release skill

### PR DESCRIPTION
**Pull request by solace-awaheed**
## Summary

  Improve the `cut-release` skill and the release-notes extraction script: tighten changelog gap-detection, make the long waits faster, publish a human-readable GitHub Release summary, and harden the release gate. Surfaced while cutting v0.6.0 (SOL-152495).

  Tracked under SOL-152495 (no GitHub issue).

  ## Type of Change

  - [x] Bug fix (non-breaking change which fixes an issue)
  - [x] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to change)
  - [x] Documentation update
  - [ ] Refactoring (no functional changes)
  - [x] Performance improvement
  - [ ] Dependency update

  ## Motivation and Context

  Cutting v0.6.0 exposed rough edges in the `cut-release` skill: gap detection over-fired (11 candidates for 1 real gap, 10 wasted `/changelog` runs), the CI/release waits had no bound and burned poll cycles, the published GitHub Release was the verbatim (very verbose) CHANGELOG block, and the release-notes gate accepted a heading-only block. This PR addresses the SOL-152495 improvement list.

  ## Changes Made

  - **Gap detection (skill A2):** a surface match is now a *candidate*, not a gap — added umbrella-entry,
  delivered-ticket-not-follow-ups, `tools.yaml` prose-only, and diff-confirmation gates so `/changelog` only runs for confirmed gaps.
  - **Empty-release guard (A2):** stop before promoting when nothing shipped since `vPREV`, with branch cleanup.
  - **UTC cutoff normalization (A2):** compare `gh` `mergedAt` (UTC) against a `date(1)`-free UTC cutoff from git, fixing a timezone string-compare bug.
  - **Faster runs (Phase C/D/E):** replaced foreground poll loops with a single `gh run watch --exit-status` and one backgrounded PR watcher; added expected durations and a `queued`-vs-swallowed-trigger distinction.
  - **Conditional `THIRD_PARTY_LICENSES` reminder; optional post-tag cleanup.**
  - **Human-readable release notes (#8):** `extract-release-notes.sh` now publishes a curated `.github/release-notes/<tag>.md` when present, else the verbatim block; skill step A6 drafts it and Phase B commits it.
  - **Strengthened release gate:** `extract-release-notes.sh` now refuses an entry-less block (previously a heading-only block passed the non-empty-file check).
  - **Bug fixes in the above:** avoided a `printf | grep -q` + `pipefail` SIGPIPE false-refusal on >64KB blocks (herestring); fixed the Phase D awk whole-record reference to `$(0)` (a literal `\$0` is an awk syntax error).

  ## Testing

  > No Go code is touched — this is a shell script + skill (Markdown) change, so `go test -race` and other Go-specific items below are **N/A**. Verified by exercising the script directly.

  **Test Environment:**
  - Go version: N/A (no Go code changed)
  - OS: macOS (darwin, bash 3.2 / BSD awk) locally; CI runs on ubuntu-latest
  - Broker version (if applicable): N/A

  **Tests Performed:**
  - [ ] Unit tests added/updated — N/A (shell/skill change)
  - [ ] Integration tests added/updated (if applicable) — N/A
  - [ ] E2E tests added/updated (if applicable) — N/A
  - [ ] Tested locally with `go test -race ./...` — N/A (no Go code)
  - [ ] Tested with real Solace broker — N/A

  **Test Coverage:**
  - `extract-release-notes.sh` exercised on: valid block → publishes; verbatim scoped to the target version; curated summary preferred when present; missing version → refuse (exit 1); heading-only block → refuse (exit 1); >64KB block → publishes (SIGPIPE regression); prefix collision (`0.1.0` vs `0.11.0`); newest block excludes `[Unreleased]`; summary file for a different tag ignored; default output filename.
  - Skill snippets: Phase D `$(0)` awk extraction on a realistic multi-version CHANGELOG; confirmed literal `\$0` is an awk syntax error and `$(0)` is equivalent to `$0` and substitution-proof.
  - Portability spot-checked on macOS bash 3.2 / BSD awk; CI target is Linux.

  ## Breaking Changes

  N/A for valid usage. Note: the release-notes gate now refuses an entry-less `## [X.Y.Z]` block that previously passed the bare non-empty-file check — this only rejects a malformed/unfinished promotion and cannot affect a correctly promoted release.

  **Impact:** None for a well-formed release.

  **Migration Guide:** None.

  ## Checklist

  ### Code Quality
  - [x] Code follows the [coding standards](../.github/CONTRIBUTING.md#coding-standards)
  - [x] Self-review completed (checked for typos, logic errors, edge cases)
  - [x] Code is well-commented (explains "why", not "what")
  - [x] Complex logic has explanatory comments
  - [x] No commented-out code or debug statements left in

  ### Security
  - [x] **No credentials in code** (checked all files, commits, and logs)
  - [ ] Followed [secure logging rules](../docs/secure-logging-rules.md) — N/A (no Go/logging changes)
  - [ ] Credential-carrying types implement `slog.LogValuer` — N/A
  - [x] No security vulnerabilities introduced (bash + Markdown only)

  ### Testing
  - [ ] All tests pass locally: `go test -race ./...` — N/A (no Go code changed)
  - [ ] No race conditions detected — N/A
  - [x] Edge cases covered (verified via direct script runs — see Test Coverage)
  - [x] Error paths tested (refusal exits verified)
  - [x] Test names clearly describe what is being tested

  ### Documentation
  - [ ] README.md updated (if user-facing changes) — N/A
  - [ ] docs/ updated (if architecture/design changes) — N/A
  - [x] godoc comments added/updated for exported functions — N/A (no Go); script/skell comments updated
  - [ ] CHANGELOG.md updated (if applicable) — N/A (skill + CI tooling, not production surface)
  - [ ] Examples updated (if applicable) — N/A

  ### Git & CI
  - [x] All commits are signed off (DCO: `git commit -s`)
  - [x] Commits have clear, descriptive messages
  - [x] Branch is up to date with main (rebased if needed)
  - [x] No merge conflicts
  - [ ] CI checks passing (lint, build, test, E2E) — will confirm after push

  ### Breaking Changes (if applicable)
  - [ ] Breaking changes documented in commit message — N/A
  - [ ] Breaking changes documented in PR description — see note above
  - [ ] Migration guide provided — N/A
  - [ ] Deprecated functionality marked with comments and deprecation notices — N/A

  ## Screenshots (if applicable)

  N/A.

  ## Additional Notes

  - The release pipeline (`release.yml`) needs no change — `extract-release-notes.sh` is a drop-in and CI still publishes
  `release-notes.md`.
  - The curated A6 summary is human-reviewed in this PR; there is no automated content check on it (an empty-summary guard was
  intentionally left out), though the CHANGELOG gate still prevents shipping a release with no notes.
  - Pre-existing awk quirk left as-is: extracting the *oldest* CHANGELOG block over-captures trailing `## Release Process`/`## Links`;
  it never triggers on a forward release.

### CODE REVIEW

File: .github/scripts/extract-release-notes.sh [hunk 1]
```diff
@@ -1,35 +1,65 @@
...
+fi
+
+# Prefer the curated summary committed in the release PR; else the verbatim block.
+if [ -f "$SUMMARY" ]; then
+  cp "$SUMMARY" "$OUT"
+  echo "Published curated release notes for ${VERSION} from ${SUMMARY} into ${OUT}." >&2
+else
+  printf '%s\n' "$BLOCK" > "$OUT"
+  echo "No ${SUMMARY}; published verbatim CHANGELOG block for ${VERSION} into ${OUT}." >&2
+fi
```
[the remark this answers is not in the document]
Comment (solace-wmaryam) [reply .github/scripts/extract-release-notes.sh:70]: I agree with Copilot here. Was there a specific reason an empty-summary guard was left out?

[the remark this answers is not in the document]
Comment (solace-awaheed) [reply .github/scripts/extract-release-notes.sh:70]: Thanks for pointing it out, missed on my end - will fix in next commit

**Review by solace-lootii — CHANGES_REQUESTED** [review]
Verified the core fixes independently and they hold up — the SIGPIPE/herestring
fix is a real bug fix (confirmed the pipefail + early-exit mechanics), the UTC
cutoff and prefix-collision handling are correct, and the `release.yml` "drop-in"
claim checks out (`GITHUB_REF_NAME` is v-prefixed, matches `SUMMARY`'s filename).

Requesting changes on one thing before merge:

**No automated test for `extract-release-notes.sh`.** This repo already has the
pattern for exactly this — `dco-check.sh` ships with `dco-check.test.sh`, wired
into `dco.yaml`, precisely so a gating script's logic is "verified on every PR
rather than trusted." This PR changes `extract-release-notes.sh`'s gating logic
(heading-only rejection, SIGPIPE fix, curated-summary preference) with only
manual verification — the 9 scenarios in the PR body's Test Coverage section are
basically already written, just not committed as a repeatable test. Given this
script is the actual release gate, I'd like those scenarios turned into an
`extract-release-notes.test.sh` mirroring `dco-check.test.sh`'s structure and
added as a CI step, same as the DCO check gets.

Non-blocking, your call:
- Phase C/D/E's "bound the wait (~30 min)" isn't enforced in the sample loop/
  `gh run watch` snippets — worth a `timeout` wrapper so it's mechanical, not
  just remembered.
- A6's curated summary is `cp`'d with no non-empty check — a one-line `[ -s
  "$SUMMARY" ]` guard would catch an accidentally-empty commit even though you
  deliberately left it out.

Nice work on the actual bug hunting here — happy to re-review once the test file's in.

**Review by solace-aross — APPROVED** [review]
Approving. Last round's blocking ask is satisfied, and the new test file is a real test rather than a rubber stamp.

I ran `extract-release-notes.test.sh` against both versions of the script:
- 9/9 pass on this branch.
- 3 fail against `main`'s script: entry-less block refuses, curated summary is published, empty summary falls back and warns. Those are exactly the three behaviours this PR adds, so the tests are coupled to the change.

Case 9 guards the new code rather than fixing something old, since the SIGPIPE hazard arrives with the entry check itself. The comment says as much, so no issue.

Two non-blocking notes.

`.claude/skills/cut-release/SKILL.md:200` claims `MERGE_SHA` "holds the `mergeCommit` oid; carry it into Phase D". It won't. The loop runs as a background command, so the variable dies with that shell, and the `MERGED` branch prints nothing. Phase D's fallback at line 209 recovers the same value from `gh pr view`, so the wrong commit never gets tagged; the sentence is just untrue on every run. Adding `echo "MERGE_SHA=$MERGE_SHA"` before the `break` would make it true.

FOSSA / SCA Scan is red, but the log shows `FOSSA_PROJECT_ID`, `FOSSA_BRANCH` and `FOSSA_REVISION` all empty with both gates skipped. That is the vault/config path failing, not a finding against this change, and this PR touches no dependencies.

**Review by solace-lootii — APPROVED** [review]
